contracts/mesh-provider/src/ibc.rs

```diff
@@ -172,6 +174,14 @@ pub fn ibc_packet_ack(
             },
             false,
         ) => fail_stake(deps, key, validator, amount),
+        (
```

Comment:
Again, this covers failure mode for unstake. What you did for unstake, you should really do for stake instead.

And for unstake, something like you do for list_validators.